assets/js/app.js

```diff
@@ -18,6 +18,14 @@ import NProgress from "nprogress"
 import { LiveSocket } from "phoenix_live_view"
 import Hooks from "./hooks"
 
+window.rememberName = () => {
+    let name = document.getElementById("name").value
+    let xhr = new XMLHttpRequest()
+    xhr.open("POST", window.location.origin + "/__api/set_name", true)
+    xhr.setRequestHeader("Content-Type", "application/json")
+    xhr.send(JSON.stringify({ name: name }))
+}
```

Comment:
Could this be done in a hook instead?

Reply:
I think hooks are generally meant for the client to react when the server does something. Here, it’s the client doing a thing. I guess we could do it with a hook, though, reacting to the update coming back when we change the Presence state. 
Or maybe I misinterpreted hooks, and they can be used to hook in on outgoing client events.

Reply:
I think hooks are generally meant for the client to react when the server does something. Here, it’s the client doing a thing. I guess we could do it with a hook, though, reacting to the update coming back when we change the Presence state. 
Or maybe I misinterpreted hooks, and they can be used to hook in on outgoing client events.

Reply:
I tried out setting up an event listened on the mounted hook. Feels reasonable, so I'll go with this.

Reply:
Yeah, I believe I've seen "official" code use hooks that way 👍